### ISSUE TITLE
Why is the supertype of AFAbstractArray 4-dimensional?

### ISSUE THREAD

**Issue by richardreeve** [JuliaComputing/ArrayFire.jl#47]
I really like the idea of using this package, but I'm slightly troubled that the ultimate supertype of all AFArrays (via AFAbstractArray) is a 4-dimensional AbstractArray if I'm reading the code correctly. This seems (to me) both counterintuitive and unhelpful in terms of handling it in a standard way. I'm sure there's a good reason for it (it might be helpful to know what it is), but is it really necessary?

As it is, I'm not sure how to generally write functions that (say) operate on matrices, which I would like to say require an AbstractArray{T, 2}, so that they can handle ArrayFire arrays too.

**Comment by ranjanan** [comment]
I have kept `AFArray` as a subtype of `AbstractArray{T,4}` because the `array` object supports at the max 4 dimensions, so I wanted to reflect that. 

I do realize what you mean: `AFArray{Float32} <: AbstractArray{Float32}`returns `true` while `AFArray{Float32,2} <: AbstractArray{Float32,2}` returns `false`

**Comment by richardreeve** [comment]
That's what I mean, yes. I would like ```AFArray{Float32,2} <: AbstractArray{Float32,2}```, so that my functions can just have arguments that are, for example
```{julia}
function coolmatrixfunction{S <: AbstractFloat}(matrix::AbstractArray{S, 2})
```
And they'll just work on any subtypes like AFArrays, because the underlying algorithms aren't meaningful anything other than matrices.

**Comment by ranjanan** [comment]
The issue was fixed in #65. You can try: 
```
julia> function thing(a::AbstractMatrix{Float32})
       end
thing (generic function with 1 method)
julia> thing(AFArray(rand(Float32, 10)))
ERROR: MethodError: no method matching thing(::ArrayFire.AFArray{Float32,1})
Closest candidates are:
  thing(::AbstractArray{Float32,2}) at REPL[3]:2

julia> thing(AFArray(rand(Float32, 10,10)))

```
Also, 
```
julia> AFArray{Float32,2} <: AbstractArray{Float32,2}
true
```

**Comment by richardreeve** [comment]
@ranjanan - excellent, thanks very much!
